utils/check_pack_uniqueness.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+const path = require('path');
+const glob = require('glob');
+const { 
+  readFile, 
+  removeCWDPrefix 
+} = require('./helpers');
+
+/**
+ * Finds the path to all top level pack configs
+ * @returns {String[]} An array of the file paths
+ */
+const findMainPackConfigFiles = () => glob.sync(path.resolve(process.cwd(), 'packs/**/config.yml')); 
+
+/**
+ * Removes whitespace and punctuation from a string
+ * @returns {String} The string with `-` replacing whitespace and punctuation removed
+ */
+const cleanPackName = (str) =>
+  str
+    .trim()
+    .toLowerCase()
+    .replace(/\s+/g, '-')
+    .replace(/-+/, '-')
+    .replace(/[^a-z0-9-]/g, '');
+
+/**
+ * Returns any packs with matching names
+ * @param {Object[]} namesAndPaths an array of objects containing the path and name of a pack
+ * @returns {Object[]|undefined} an array of matching values or undefined if there are none
+ */
+const getMatchingNames = (namesAndPaths) => {
+  const nameSet = new Set();
+  for (const p of namesAndPaths) {
+    if (nameSet.has(p.name)) {
+      return namesAndPaths.filter(pack => pack.name === p.name);
```

Comment:
i think it might be worth returning all matching tuples, rather than the first matching tuple.

Reply:
I updated the logic to search for all matches before returning.